pkg/reconciler/v1alpha1/revision/reconcile_resources.go

```diff
@@ -229,7 +229,7 @@ func (c *Reconciler) reconcileService(ctx context.Context, rev *v1alpha1.Revisio
 		// If the endpoints resource is NOT ready, then check whether it is taking unreasonably
 		// long to become ready and if so mark our revision as having timed out waiting
 		// for the Service to become ready.
-		revisionAge := time.Since(getRevisionLastTransitionTime(rev))
+		revisionAge := time.Since(endpoints.CreationTimestamp.Time)
```

Comment:
I do not think this is correct. Since, if I have an inactive revision the endpoints resource is still there. So when reactivation happens and if it takes some time to schedule the pod, this will fail the revision.

Reply:
When the Regision is inactivate we cannot hit this because `IsActivationRequired`.  We should literally only hit this during the initial deployment.